programs/stake-tests/tests/test_move_stake_and_lamports.rs

```diff
@@ -0,0 +1,1281 @@
+        program_test.deactivate_feature(*feature_id);
+    }
+
+    program_test
+}
+
+#[derive(Debug, PartialEq)]
+struct Accounts {
+    validator: Keypair,
+    voter: Keypair,
+    withdrawer: Keypair,
+    vote_account: Keypair,
+}
+
+impl Accounts {
+    async fn initialize(&self, context: &mut ProgramTestContext) {
+        let slot = context.genesis_config().epoch_schedule.first_normal_slot + 1;
+        context.warp_to_slot(slot).unwrap();
+
+        create_vote(
+            context,
+            &self.validator,
+            &self.voter.pubkey(),
+            &self.withdrawer.pubkey(),
+            &self.vote_account,
+        )
+        .await;
+    }
+}
+
+impl Default for Accounts {
+    fn default() -> Self {
+        let vote_account = Keypair::new();
+
+        Self {
+            validator: Keypair::new(),
+            voter: Keypair::new(),
+            withdrawer: Keypair::new(),
+            vote_account,
+        }
+    }
+}
+
+async fn create_vote(
+    context: &mut ProgramTestContext,
+    validator: &Keypair,
+    voter: &Pubkey,
+    withdrawer: &Pubkey,
+    vote_account: &Keypair,
+) {
+    let rent = context.banks_client.get_rent().await.unwrap();
+    let rent_voter = rent.minimum_balance(VoteState::size_of());
+
+    let mut instructions = vec![system_instruction::create_account(
+        &context.payer.pubkey(),
+        &validator.pubkey(),
+        rent.minimum_balance(0),
+        0,
+        &system_program::id(),
+    )];
+    instructions.append(&mut vote_instruction::create_account_with_config(
+        &context.payer.pubkey(),
+        &vote_account.pubkey(),
+        &VoteInit {
+            node_pubkey: validator.pubkey(),
+            authorized_voter: *voter,
+            authorized_withdrawer: *withdrawer,
+            ..VoteInit::default()
+        },
+        rent_voter,
+        vote_instruction::CreateVoteAccountConfig {
+            space: VoteStateVersions::vote_state_size_of(true) as u64,
+            ..Default::default()
+        },
+    ));
+
+    let transaction = Transaction::new_signed_with_payer(
+        &instructions,
+        Some(&context.payer.pubkey()),
+        &[validator, vote_account, &context.payer],
+        context.last_blockhash,
+    );
+
+    // ignore errors for idempotency
+    let _ = context.banks_client.process_transaction(transaction).await;
+}
+
+async fn transfer(context: &mut ProgramTestContext, recipient: &Pubkey, amount: u64) {
+    let transaction = Transaction::new_signed_with_payer(
+        &[system_instruction::transfer(
+            &context.payer.pubkey(),
+            recipient,
+            amount,
+        )],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap();
+}
+
+async fn advance_epoch(context: &mut ProgramTestContext) {
+    refresh_blockhash(context).await;
```

Comment:
ive never had issues with test flakiness (if you mean tests that sometimes pass sometimes fail), i added this as a convenience for some more complicated neostake tests where i would run transactions with the same `Message`, because i kept forgetting i needed a new blockhash for it to not be flagged as duplicate

Reply:
ah gotcha, thanks for the info! I'll give it a shot if I run into warping issues again